### DISCUSSION TITLE
Frigate LXC

### DISCUSSION THREAD

**Discussion by ecard0** [tteck/Proxmox#197] [Q&A]
This would be a great addition to the list :)     Thanks for the great work

**Comment by carlreid** [comment]
Not sure how the [frigate-support.sh](https://github.com/tteck/Proxmox/blob/main/misc/frigate-support.sh) plays into this, but this wold also be a great addition for me too

**Reply by maxwood** [reply]
Any idea what frigate-support.sh does?

**Reply by tteck** [reply]
https://github.com/tteck/Proxmox/blob/a85593afa43095c4bd527748af7090e3e4840311/misc/frigate-support.sh#L8

**Reply by maxwood** [reply]
I'm new to all this, just trying to understand but the script is like a foreign language to me. Also, do I need to setup an LXC or Docker LXC beforehand?

**Reply by tteck** [reply]
Yes, usually a Docker LXC. If you're interested in Frigate, I would suggest browsing the [user-submitted guides](https://github.com/tteck/Proxmox/blob/main/USER_SUBMITTED_GUIDES.md) to find a solution.

**Reply by tteck** [reply]
I recommend checking out Scrypted as an alternative to Frigate.

**Reply by maxwood** [reply]
Thanks, I’ve got it all setup now though, I’ve been using Frigate for a while but as a Home Assistant addon. I wanted to move it to a separate container, it seems to be working but if you get 5 mins to explain what the frigate-support.sh script is doing I’d really appreciate it! :)

**Reply by tteck** [reply]
It's an autodev hook script that configures an array of device types to be enabled within a container.

**Reply by tteck** [reply]
@maxwood did you utilize the `frigate-support.sh` script?

**Reply by tteck** [reply]
Updated `frigate-support.sh` to include descriptions, along with a menu that enables you to choose the specific container where you want to add support.

**Reply by maxwood** [reply]
Yes I did, I used your LXC Docker script and the Frigate Support script. 

I added these two lines `lxc.mount.entry: /dev/dri/renderD128 dev/dri/renderD128 none bind,optional,create=file 0, 0
lxc.mount.entry: /dev/bus/usb/002/ dev/bus/usb/002/ none bind,optional,create=dir 0,0` to the lxc config file based on this tutorial https://www.homeautomationguy.io/blog/running-frigate-on-proxmox

But I'm not sure if it was necessary and the frigate support script covered something like this already.

**Comment by es3nric** [comment]
Same here, I would be great to have

**Comment by DocN0** [comment]
that would be great, are there already any approaches or something where we could support?

**Comment by timdonovanuk** [comment]
+1. Being able to pass through GPU would be great.

**Comment by Nindohana** [comment]
+1 would be absolutely awesome if a frigate script could be created.

**Comment by bam27beer** [comment]
have my coral waiting to deploy ,just trying to find the time to learn to install

**Comment by tteck** [comment]
https://github.com/blakeblackshear/frigate/discussions/1111

**Comment by kornelremete** [comment]
it would be good to have

**Comment by ncodee** [comment]
+1, hopefully will be added soon.

**Comment by petrolette** [comment]
After installing Frigate LXC, the console asks me for a login and password.
Who can help me?

**Reply by DrD00der** [reply]
If you didn't figure this out yet, it's usually root/your default password for proxmox.  If you do an advanced installation you can set the root password to whatever you want

**Comment by DrD00der** [comment]
Getting an error when trying to run the script: 

[ERROR] in line 17: exit code 0: while executing command $STD apt-get install -y {curl,sudo,mc,git,gpg,automake,build-essential,xz-utils,libtool,ccache,pkg-config,libgtk-3-dev,libavcodec-dev,libavformat-dev,libswscale-dev,libv4l-dev,libxvidcore-dev,libx264-dev,libjpeg-dev,libpng-dev,libtiff-dev,gfortran,openexr,libatlas-base-dev,libssl-dev,libtbb2,libtbb-dev,libdc1394-22-dev,libopenexr-dev,libgstreamer-plugins-base1.0-dev,libgstreamer1.0-dev,gcc,gfortran,libopenblas-dev,liblapack-dev,libusb-1.0-0-dev,jq}

Any suggestions?  Ran it verbose mode, tried it both privileged and unprivileged.  Used Debian as suggested above

**Reply by LinkXX92** [reply]
i have same problem here :(

EDIT: Use debian 11

**Comment by revnyolc** [comment]
[ERROR] in line 23: exit code 0: while executing command "$@" > /dev/null 2>&1
The silent function has suppressed the error, run the script with verbose mode enabled, which will provide more detailed output.

**Reply by tteck** [reply]
🛑 Failure to comply with the following guideline.
> - When encountering the error message [ERROR] in line 23: exit code *: while executing command "$@" > /dev/null 2>&1, make sure to run the script in verbose mode to accurately determine the underlying issue.

**Comment by miketrump** [comment]
Hello @tteck and thank you so much for this great work !
there is a new update (0.14.0) relesed 2 weeks ago.
How to update to it?
my current version is 0.13.2.
I did try (update) command in frigate lxc and get this message (There is currently no update path available).
Do we have to wait? or am I doing it wrong?

**Reply by hajobu** [reply]
same here
if I do a new install it runs frigate 0.14.0

**Reply by danbutter** [reply]
I'm on 0.14 and was wanting to upgrade to .14.1 so I was trying to figure this out also. As mentioned above some of the scripts say to just type update in the LXC shell so I tried that today and got:

✗ To update Frigate, create a new container and transfer your configuration.

So while that method will work it isn't an easy way to upgrade to latest.  If there is an easier way I haven't found it yet.
